src/arch/cortex-m0/syscalls.S

```diff
@@ -0,0 +1,34 @@
+.cpu cortex-m0
+.syntax unified
+.thumb
+.text
+
+.section .syscalls
+
+/* Cortex-M0 can only push/pop registers R0-R7 directly, so move R8-R11 to/from
+ * R0-R3. This is equivalent to the 32-bit "push/pop {r4-r11}" instructions. */
+
+.macro SYSCALL NAME, NUM
+.global \NAME
+.thumb_func
+\NAME :
```

Comment:
I used this porting task for testing these fancy features :) I can do that once I confirm that __wait() does indeed not need to save/restore LR (did you test removing it from the push/pop call for storm?)